runtime-modules/membership/src/lib.rs

```diff
@@ -741,6 +784,153 @@ decl_module! {
+                Error::<T>::InsufficientBalanceToGift
+            );
+
+            ensure!(
+                gifter_free_balance >= total_credit,
+                Error::<T>::InsufficientBalanceToGift
+            );
+            ensure!(
+                balances::Pallet::<T>::can_slash(&gifter, membership_fee),
+                Error::<T>::InsufficientBalanceToGift
+            );
+
+            if let Some(locked) = params.apply_controller_account_invitation_lock {
+                ensure!(
+                    params.credit_controller_account >= locked,
+                    Error::<T>::GifLockExceedsCredit
+                );
+            };
+
+            if let Some(locked) = params.apply_root_account_invitation_lock {
+                ensure!(
+                    params.credit_root_account >= locked,
+                    Error::<T>::GifLockExceedsCredit
+                );
+            };
+
+            // Check for existing invitation locks on controller account.
+            ensure!(
+                T::InvitedMemberStakingHandler::is_account_free_of_conflicting_stakes(
+                    &params.controller_account
+                ),
+                Error::<T>::ConflictingLock,
+            );
+
+            // Check for existing invitation locks on root account if not same
+            // as controller account.
+            if params.root_account != params.controller_account {
+                ensure!(
+                    T::InvitedMemberStakingHandler::is_account_free_of_conflicting_stakes(
+                        &params.root_account
+                    ),
+                    Error::<T>::ConflictingLock,
+                );
+            }
+
+            //
+            // == MUTATION SAFE ==
+            //
+
+            // Transfer funds to controller account
+            <balances::Pallet::<T> as Currency<T::AccountId>>::transfer(
+                &gifter,
+                &params.controller_account,
+                params.credit_controller_account,
+                ExistenceRequirement::AllowDeath
+            )?;
+
+            // Transfer funds to root account
+            <balances::Pallet::<T> as Currency<T::AccountId>>::transfer(
+                &gifter,
+                &params.root_account,
+                params.credit_root_account,
+                ExistenceRequirement::AllowDeath
+            )?;
+
+            // slash fee, balance_not_slashed should be zero
+            let (_negative_imbalance, balance_not_slashed) = balances::Pallet::<T>::slash(
+                &gifter,
+                membership_fee
+            );
+
+            // Ensure the entire fee was slashed. This should not fail
+            // since we checked for sufficient usable balance.
+            ensure!(
+                balance_not_slashed == Zero::zero(),
+                Error::<T>::InsufficientBalanceToGift
+            );
+
+            if params.root_account != params.controller_account {
```

Comment:
This is, technically speaking, not necessary, since locks overlay each other and don't stack. 
So adding the another lock with the same amount won't change the way locks are accounted for

Reply:
Correct. Here it was just an optimization so if we already checked once, we don't need to check a second time if the accounts are the same.